test/unit/logger/logger.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include <iostream>
+
+#include <gtest/gtest.h>
+
+#include "logger/ur_logger.h"
+
+TEST(logger, SetLevel) {
+    auto level = level::Level::WARNING;
+    auto sink = new sink::Sink();
```

Comment:
Do not use unsafe code because you can. :-)
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-newdelete